notes/jinfwhuang/beacon-chain-light-client.md

```diff
@@ -0,0 +1,60 @@
+# Beacon Chain Light Client
+
+## My current understanding
+
+#### Protocol specs
+
+1. Consensus protocol and execution
+
+- The consensus protocol as of Altair already supports light client state updates. See [sync-protocol](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/sync-protocol.md) and [beacon-chain](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/beacon-chain.md)
+- There are some discussions on how light clients use state root and merkle proofs to create transactions, but they are not in the specs.
+
+2. Network protocol
+
+- It doesn't look like that there is any design decisions made on how light client fetch block headers, committee signatures, and merkle proofs (witnesses).
+- Two basic approaches: talk to full notes, talk to a p2p light client subnet
+- There seems to be an overlap with **portal network** on how to allow light weight clients to get the data that they need.
+
+#### Client Implementations
+
+- Most Beacon chain implementations have not implemented the "state update" logic.
+
+- A note on some implementation details on the Lodestar Beacon Chain: https://hackmd.io/hsCz1G3BTyiwwJtjT4pe2Q
+
+- The v2 release of Prysm, which is compatible with Altair hard fork, only implements sync committee logic that is required by the consensus protocol. It has not implemented any state update logic. This could be a good POC project to work on.
+
+## My next steps
+
+- Find a couple of people who is actively thinking about this topic. I want to have a chat with them with respect to the following questions.
+  - Is there a plan to motivate client implementations to have a light client option? What is the timing of that plan?
+  - With regard to Beacon Chain light clients, what is the "spec plan" and "implementation plan"?
+  - What is the relationship between portal network and light client? Should light client be participating on the portal network? Is light client equivalent to stateless client? Will they co-exist or there going to be only one of them?
```

Comment:
@pipermerriam I want to ask for your help on identifying a knowledgable person to guide me to get up-to-date information on these questions. Do you have a free cycle to help?

Reply:
I think you've already found it but asking these types of things in the Portal Network discord is a good place.